hansard/uk_to_dolma.py

```diff
@@ -0,0 +1,164 @@
+
+FILE_NAMES = {
+    "debates": {"source": "commons-debates"},
+    "london-mayors-questions": {
+        "source": "london-mayors-questions",
+    },
+    "lordspages": {"source": "lords-debates"},
+    "lordswms": {
+        "source": "lords-written-ministerial-statements",
+    },
+    "lordswrans": {
+        "source": "lords-written-answers",
+    },
+    "ni": {"source": "northern-ireland-assembly"},
+    "sp": {"source": "scottish-parliament"},
+    "sp-written": {
+        "source": "scottish-parliament-written-answers",
+    },
+    "standing": {"source": "standing-committees"},
+    "westminister": {"source": "westminister-hall"},
+    "wms": {"source": "written-ministerial-statements"},
+    "wrans": {"source": "written-answers"},
+    "senedd": {
+        "cy": {"source": "senedd-cy"},
+        "en": {"source": "senedd-en"},
+        "source": "senedd",
+    },
+}
+WHITESPACE = regex.compile(r"\w+|[^\w\s]+")
+COUNT = 0
+
+PARSER = lxml.etree.XMLParser(
+    encoding="utf-8",
+    recover=True,
+)
+
+
+def get_subfolders(folder_path: str | Path) -> list[Path]:
+    if isinstance(folder_path, str):
+        folder_path = Path(folder_path)
+    subfolders = [f for f in folder_path.iterdir() if f.is_dir()]
+    return subfolders
+
+
+def parse_hansard_xml_file(root: ET._Element) -> str:
+    parsed_text = []
+    for element in root.iter():
+        if element.tag == "major-heading":
+            text = ET.tostring(element, method="text", encoding="unicode").strip()
+            parsed_text.append(f"{text}")
+        elif element.tag in ["oral-heading", "minor-heading"]:
+            text = ET.tostring(element, method="text", encoding="unicode").strip()
+            parsed_text.append(f"{text}")
+        elif element.tag in [
+            "speech",
+            "ques",
+            "reply",
+            "question",
+        ]:
+            speaker = (
+                element.attrib.get("speakername", "") + ": "
+                if element.attrib.get("speakername", "")
+                else ""
+            )
+            speech_text = (
+                ET.tostring(element, method="text", encoding="unicode")
+                .strip()
+                .replace("    ", "")
+            )
+            parsed_text.append(f"{speaker + speech_text}")
+
+    return "\n\n".join(parsed_text).replace(" ", "")
+
+
+def process_files_in_folder(
+    folder_path: Path, source: str, count: bool = False
+) -> dict:
+    global COUNT
+    language = "en" if not source == "senedd-cy" else "cy"
+    date_match = re.compile(r"\d{4}-\d{2}-\d{2}")
+    for file in folder_path.iterdir():
+        if file.suffix == ".xml" and file.stem != "tmp":
+            root = ET.parse(file, parser=PARSER).getroot()
+            parsed_text = parse_hansard_xml_file(root)
+            if parsed_text:
+                if count:
+                    COUNT += len(WHITESPACE.split(parsed_text))
+                date_ = date_match.search(file.stem)
```

Comment:
This is one of the best places for the walrus operator:

```python
date = d.group() if (d := date_match.search(file.stem)) else None
```

Also, I think `None` is a better default than a year far in the future